src/Workspaces/Core/Portable/Workspace/Solution/ProjectId.cs

```diff
@@ -24,6 +24,11 @@ namespace Microsoft.CodeAnalysis
     [DataContract]
     public sealed class ProjectId : IEquatable<ProjectId>
     {
+        /// <summary>
+        /// Checksum of this ProjectId, built from our <see cref="Id"/> and <see cref="DebugName"/>.
+        /// </summary>
```

Comment:
For purposes of equality, only the ID matters -- so this would mean you could have two ProjectId that are equal but have different checksums. I could imagine that might causes surprises somewhere. (I'd also hope somewhere we might have a helper for just GUID to checksum that is fast since it'd just be a matter of copying some bits?)

Reply:
I'm not going to hit signoff pending this question -- I'm not sure it matters but if this is one of those "checksums don't behave right" we might get very strange behavior.

Reply:
oh yikes.  this is a very good call.  i will fix *and* test for this.

Reply:
fixed, with tests.